src/Compilation.zig

```diff
@@ -251,7 +251,12 @@ reference_trace: ?u32 = null,
 libcxx_abi_version: libcxx.AbiVersion = libcxx.AbiVersion.default,
 
 /// This mutex guards all `Compilation` mutable state.
-mutex: std.Thread.Mutex = .{},
+/// Disabled in single-threaded mode because the thread pool spawns in the same thread.
+mutex: if (builtin.single_threaded) struct {
+    pub inline fn tryLock(_: @This()) void {}
+    pub inline fn lock(_: @This()) void {}
+    pub inline fn unlock(_: @This()) void {}
+} else std.Thread.Mutex = .{},
```

Comment:
was this needed? std.Thread.Mutex already checks builtin.single_threaded

Reply:
It does, but a non-recursive mutex does not function in a single-threaded compiler because the code calls `comp.thread_pool.spawn` with the mutex locked which synchronously executes code that needs to lock the mutex.  I considered making it a recursive mutex, but that would just do pointless work in threaded mode, and even more pointless work in single threaded mode.